sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -1122,16 +1122,26 @@ def test_gradient_boosting_early_stopping():
     )
 
     X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
-    # Check if early_stopping works as expected
-    for est, tol, early_stop_n_estimators in (
+    # Check if early_stopping works as expected, that is empirically check that the
+    # number of estimators is increasing when the tolerance decreases.
+
+    # Depending on platforms, the number of fitted estimators might slightly vary.
+    # Hence, we check for its inclusion in an interval centered in the expected
+    # number of fitted estimators rather than a strict equality.
+    delta_early_stop_n_estimators = 2
+    for est, tol, expected_early_stop_n_estimators in (
         (gbc, 1e-1, 28),
         (gbr, 1e-1, 13),
         (gbc, 1e-3, 70),
         (gbr, 1e-3, 28),
     ):
         est.set_params(tol=tol)
         est.fit(X_train, y_train)
-        assert est.n_estimators_ == early_stop_n_estimators
+        assert (
+            expected_early_stop_n_estimators - delta_early_stop_n_estimators
```

Comment:
I think it would be enough to have an upper bound with a less-than-operator.

Reply:
What I don't get here is why Python 3.11 would trigger this issue while other Windows builds are working.

Reply:
can we just follow the description of the test and simply check the following ?
```gbc(tol=1e-1).fit(X, y).n_estimators_ < gbc(tol=1e-3).fit(X, y).n_estimators_``

Reply:
Done in c47bbbdc77fc0308c5126f05d67f56600303c538.